downloads/install.sh

```diff
@@ -112,6 +112,18 @@ fi
 FILE="spin-${VERSION}-${OS}-${ARC}.tar.gz"
 URL="https://github.com/fermyon/spin/releases/download/${VERSION}/${FILE}"
 
+# Establish the location of current working environment
+current_dir=$(pwd)
+fancy_print 0 "Current working directory: ${current_dir}"
+
+# Establish the absolute Spin installation location
+spin_installation_location=${current_dir}"/spin"
+
+# Check to see if an item called spin already exists (before expending resources downloading the file).
+if [ -e "$spin_installation_location" ]; then
```

Comment:
I believe this is only an issue if it's a directory - if existing files are a problem then we should also check for `LICENSE`, `README.md` and the other files in the tarball.  (I mean... we could check if there are directories with those names too... but that seems a lot less likely than there being files.)

Reply:
Good point, indeed. I have updated in the next push.

Reply:
The script now checks for the additional items (not just `spin`), we are using `items` as not to expend additional syntax discerning if things are files or directories. If it exists, then the installer will exit.